docs/advanced-features/middleware.md

```diff
@@ -1,95 +1,166 @@
 ---
-description: Learn how to use Middleware in Next.js to run code before a request is completed.
+description: Learn how to use Middleware to run code before a request is completed.
 ---
 
-# Middleware (Beta)
+# Middleware
 
 <details open>
-  <summary><b>Version History</b></summary>
+<summary><b>Version History</b></summary>
 
-| Version   | Changes                                                                                                 |
-| --------- | ------------------------------------------------------------------------------------------------------- |
-| `canary`  | Preparing for stability, see [upgrade guide](https://nextjs.org/docs/messages/middleware-upgrade-guide) |
-| `v12.0.9` | Enforce absolute URLs in Edge Runtime ([PR](https://github.com/vercel/next.js/pull/33410))              |
-| `v12.0.0` | Middleware (Beta) added.                                                                                |
+| Version | Changes |
+
+| --------- | ------------------------------------------------------------------------------------------ |
+
+| `v12.2.0` | Middleware is stable |
+
+| `v12.0.9` | Enforce absolute URLs in Edge Runtime ([PR](https://github.com/vercel/next.js/pull/33410)) |
+
+| `v12.0.0` | Middleware (Beta) added |
 
 </details>
 
-Middleware enables you to run code before a request is completed. Based on the user's incoming request, you can modify the response by rewriting, redirecting, adding headers, or even streaming HTML.
+Middleware allows you to run code before a request is completed, then based on the incoming request, you can modify the response by rewriting, redirecting, adding headers, or setting cookies.
 
-## Usage
+Middleware runs _before_ cached content, so you can personalize static files and pages. Common examples of Middleware would be authentication, A/B testing, localized pages, bot protection, and more.
 
-1. Install the canary version of Next.js:
+> **Note:** If you were using Middleware prior to `12.2`, please see the [upgrade guide](https://nextjs.org/docs/messages/middleware-upgrade-guide).
 
-```jsx
-npm install next@canary
-```
+## Using Middleware
 
-2. Then, create a `middleware.ts` file under your project root directory.
+To begin using Middleware, follow the steps below:
 
-3. Finally, export a middleware function from the `middleware.ts` file.
+1. Install the latest version of Next.js:
 
-```jsx
+```bash
+npm install next@latest
+```
+
+2. Create a `middleware.ts` (or `.js`) file under your project root directory.
```

Comment:
```suggestion
2. Create a `middleware.ts` (or `.js`) file next to your `pages` folder.
```

Reply:
I think this phrasing is maybe more confusing than "at the project root"

Reply:
@jescalan This was addressing some feedback we'd got from users: https://vercel.slack.com/archives/C0289CGVAR2/p1656348479861419 Maybe we can be clearer in the language we're using though.

Reply:
Updated the wording in this [commit](https://github.com/vercel/next.js/pull/37992/commits/40806297a3112965814941945ae086f917761b56)